api/src/main/java/org/apache/iceberg/ContentFile.java

```diff
@@ -29,6 +29,11 @@
  * @param <F> the concrete Java class of a ContentFile instance.
  */
 public interface ContentFile<F> {
+  /**
+   * Returns the ordinal position of the file in a manifest, or null if it was not read from a manifest.
+   */
+  Long pos();
```

Comment:
qq: do we want to include anything in the name to indicate that it is a position in the manifest?

Reply:
I'd be open to that, but I couldn't come up with a very good name for this field so I just used "pos" because we use that elsewhere to indicate position in a file.